src/book/03-guides/04-general/04-documentation.mdx

```diff
@@ -0,0 +1,88 @@
+---
+section: Guides
+chapter: General
+title: Documentation
+description: How to properly document code.
+slug: /guides/general/documentation
+---
+
+## Conventions
+
+We use the javadoc documentation _style_, all this means is that directives are formatted with a `@`. Note that `@` and `\\` are interchangeable when looking at other sources.
+
+Each declaration, or definition in the absence of declaration, that is a part of an api, e.g. declared in the `.hpp` file, should have a documenting comment.
+
+### Directives
+
+This is a list of directives some that we use.
+
+- `@brief` A brief description of the following code. Every documenting comment should have this.
+- `@detail` All the details that someone using this thing might need to know beyond what is provided in other directives.
+- `@param[direction] <formal name>` Details a parameter that is passed to a function. Direction is either `in` or `out` and is usually omitted.
+- `@return` Details what a function returns.
+- `@retval <value>` Details specific return value meanings, usually for enums, booleans or error codes.
+- `@throws <type name>` Details on what a function throws.
+- `@tparam <formal name>` Details on the template parameter, usually a [named requirement](https://en.cppreference.com/w/cpp/named_req) or other requirements that must be fulfilled.
+
+This is a list of extra things that may be used, see [doxygen](https://www.doxygen.nl/manual/commands.html#cmd_intro) documentation for the full list.
+
+- `@example <file-name>` Indicates that an example of how the function or class should be used is at `<file-name>`.
+- `@see <name>` Tells the reader to read documentation for another class, function or variable.
+- `@attention` Tells the reader that they should pay attention to the following point.
+- `@warning` Tells the reader something that may go wrong, and ways of avoiding such a fate.
+- `@note` Some extra "Spicy" info which does not fit in other tags.
+- `@remark` An aside or candid comment about the code or approach.
+- `@todo` Adds a todo to the documentation.
+- `@post` Details a post condition.
+- `@pre` Details a pre condition.
+- `@code{.<language>}` and `@endcode` starts and finishes a code block respectively, like \` and in NUbook. `{.language}` is optional.
+- `@copydoc <name>` Copies the documentation from another thing.
+- `@f$`, `@f[`, `@f]`, `@f{<environment>}{` and `@f}` Creates latex mathematics.
+- `@verbatim` and `@endverbatim` Starts and finishes verbatim output respectively, outputs exactly what you wrote.
+- `@author` The author of this code. This is used sporadically in `NUbots`. We generally rely on the git blame instead.
```

Comment:
I feel like relying on git blame is dodgy. Someone used that the other week to say that I was responsible for a piece of code when I had nothing to do with it, it was formatting or something like that.